ledger/src/bank_forks_utils.rs

```diff
@@ -97,7 +97,6 @@ pub fn load_bank_forks(
             "Initializing bank snapshot path: {}",
             snapshot_config.bank_snapshots_dir.display()
         );
-        let _ = fs::remove_dir_all(&snapshot_config.bank_snapshots_dir);
```

Comment:
Because removing snapshots is no longer simply deleting the top level directory.  Doing that leaving the account_path/snapshot/ orphaned.

Reply:
The full logic will be:
  Before booting from snapshot_dir, purge all PRE snapshot dirs, purge POST snapshot dirs (retain 1);
  Before booting from snapshot_archive, purge all snapshot dirs.
  
The snapshot_dir part will come in from https://github.com/solana-labs/solana/pull/30978.  Here is only the snapshot_archive part.

To use account_paths and operate at paths level to achieve the same result is complicated (because of the hardlink associations), and duplicates the same logic as purging function already does.

Reply:
> Is this correct: If we remove the bank snapshots dir, then we'd remove the symlinks to the `accounts/snapshot/` and thus be unable to cleanup any of those directories?

There is a call of clean_orphaned_account_snapshot_dirs which can clean up the <account_path>/snapshot/\<slot\> directories if they are not pointed by snapshot_dirs.  That calling happens at earlier time together with the account run path cleaning.